sharding/shardset.go

```diff
@@ -93,9 +93,9 @@ func validateShards(shards []uint32) error {
 	return nil
 }
 
-// DefaultHashGen generates a HashFn based on murmur32
+// DefaultHashGen generates a HashFn based on ID's internal hash.
 func DefaultHashGen(length int) HashFn {
-	return func(id string) uint32 {
-		return murmur3.Sum32([]byte(id)) % uint32(length)
+	return func(id ts.ID) uint32 {
+		return uint32(id.Hash().Fold() % uint64(length))
```

Comment:
Hm not sure this is a good idea, if we use the same hash function for map keys as we do for distributing our sharded data that's not a great idea.  The two hash functions have different goals.  I'm worried that we won't get as good distribution for our data shards potentially.

Can't we just do `murmur3.Sum32(id.Data())` here?